app/services/imports/scheme_location_import_service.rb

```diff
@@ -0,0 +1,166 @@
+module Imports
+  class SchemeLocationImportService < ImportService
+    def create_scheme_locations(folder)
+      import_from(folder, :create_scheme_location)
+    end
+
+    def create_scheme_location(xml_document)
+      management_group = location_field_value(xml_document, "mgmtgroup")
+      schemes = Scheme.where(old_id: management_group)
+      raise "Scheme not found with legacy ID #{management_group}" if schemes.empty?
+
+      if schemes.size == 1 && schemes.first&.locations&.empty?
```

Comment:
Do you need the first `&.` here? Surely if `schemes.size == 1` we know we have a scheme value?

Reply:
Good point